Jint/Engine.cs

```diff
@@ -61,6 +61,13 @@ public sealed partial class Engine : IDisposable
         internal Intrinsics _originalIntrinsics = null!;
         internal Host _host = null!;
 
+        public delegate void ParsedHandler(object sender, string source, Script script);
+
+        /// <summary>
+        /// The Parsed event is triggered whenever a script has been parsed as a result of execution or evaluation.
+        /// </summary>
+        public event ParsedHandler? Parsed;
```

Comment:
nit: kind of more like `ScriptParsed` as doesn't support module code

Reply:
should these be in options? usually anything configurable is found there

Reply:
> nit: kind of more like `ScriptParsed` as doesn't support module code

I was actually writing this while you replied:

Rethinking the `Parsed` event - and the recent `Loaded` event on `DefaultModuleLoader`. Ideally (especially for the VS Code DebugAdapter), those two should be combined into just `Parsed` on `Engine`. A debugger would need at least the `source` (ID) of the parsed script - to use as for breakpoints and as a general key - as well as the AST. But of course, after parsing, the `source` is already available on the AST's root node. DebugAdapter needs the `Parsed` event (or possibly more accurately a "script ready to execute" event) in order to wait for the debugging client (e.g. VS Code) to set e.g. breakpoint positions (after which VS Code sends `configurationDone` and the script can start evaluation).

So, a more general approach would be to remove `Loaded` from `DefaultModuleLoader` again, and instead let the engine handle it for *all* module loaders, by triggering the same `Parsed` event as a "classic" script `Execute` (with arguments `object sender` and `Program ast`) here:

https://github.com/sebastienros/jint/blob/main/Jint/Engine.Modules.cs#L44

(or does it make more sense to add it in `ImportModule`?)

... and, for "classic" scripts, somewhere... inhere:

https://github.com/sebastienros/jint/blob/4b6f44596bac1eb98a9492f7a8e60de5929c820d/Jint/Engine.cs#L295

... which would "unify" `Parsed` for the two methods of loading scripts. For the case of modules, e.g. a DebugAdapter would only listen for the first time the event is triggered.

Any idea at which points the event would make most sense to add? Considering the `Parsed` event will, in the case of e.g. a DebugAdapter, be used to pause execution of the `Engine` thread - e.g. (in the VS Code example) until the debugging client is done setting up evaluation.

Reply:
> should these be in options? usually anything configurable is found there

In the original draft from last year (which I ended up deleting), they were actually calls to `DebugHandler.OnScriptReady`, and the `ScriptReady` event was on `DebugHandler` - along with `Step/Break/Skip`. Not sure about that, since I guess they may be useful outside of debugging too. How would we handle events with the `Options` fluent API?

Reply:
I'm good if you're good with the solution 😉

 I wouldn't mind if fluent API isn't there for more special options. It's Options patterns more now with fluent available for common cases IMO. There are already many switches not visible on fluent side of I remember right.

Reply:
Problem is, if we start introducing C# events to `Options`, that would require `Options` to also include logic for dispatching those events, or at the very least have an internal `OnParsed` on `Options`. Not just from a pattern perspective - that's enforced by the compiler.

Reply:
Moved `Parsed` to `DebugHandler` for now, pending this discussion - just to implement the event for modules too - and test it with the DebugAdapter as shown above.